### PR TITLE
[WFLY-11570][WFLY-11687][WFLY-11607] component upgrades to fix remote transaction propagation issues

### PR DESCRIPTION
**Pull request by ochaloup**
https://issues.jboss.org/browse/WFLY-11687
~~https://issues.jboss.org/browse/WFLY-11684~~
https://issues.jboss.org/browse/WFLY-11570
https://issues.jboss.org/browse/WFLY-11607

project issue: https://issues.jboss.org/browse/JBEAP-16089

The fix for the transaction timeout propagation error was done in two components (jboss ejb client and the wfly transaction client). That's why the component upgrades.
The reason of the changes in the integration testsuite is the different handling of timeouted transaction in wftc as it was so far. For more details please check the discussion at https://issues.jboss.org/browse/WFTC-54?focusedCommentId=13677791#comment-13677791

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java [hunk 1]
```diff
@@ -41,6 +45,8 @@
     public static final String REPLY_QUEUE_NAME = "replyQueue";
     public static final String REPLY_QUEUE_JNDI_NAME = "queue/" + REPLY_QUEUE_NAME;
 
+    ModelNode amqServerDefaultAdress = new ModelNode().add(ClientConstants.SUBSYSTEM, "messaging-activemq")
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java]: This should be a private constant or just created in the needed methods.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java [hunk 2]
```diff
@@ -63,10 +72,32 @@ public void tearDown(ManagementClient managementClient, String containerId) thro
...
+                removeAddressSettings(PROPERTY_TIMEOUT_QUEUE_NAME);
             } finally {
                 adminOperations.close();
             }
         }
     }
 
+    // /subsystem=messaging-activemq/server=default/address-setting=queue.propertyTimeoutQueue:add(max-delivery-attempts=1)
+    private void setMaxDeliveryAttempts(String queueName) throws IOException {
+        ModelNode address = amqServerDefaultAdress.add("address-setting", "queue." + queueName);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java]: This add method mutates the target ModelNode, so you need to clone it before adding if you're using a shared object.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java [hunk 3]
```diff
@@ -63,10 +72,32 @@ public void tearDown(ManagementClient managementClient, String containerId) thro
...
+
+        ModelNode operation = new ModelNode()
+            .get(ClientConstants.OP_ADDR).set(address)
+            .get(ClientConstants.OP).set(ClientConstants.ADD)
+            .get("max-delivery-attempts").set(1);
+        adminOperations.getControllerClient().execute(operation);
+    }
+
+    private void removeAddressSettings(String queueName) throws IOException {
+        ModelNode address = amqServerDefaultAdress.add("address-setting", "queue." + queueName);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java]: This add method mutates the target ModelNode, so you need to clone it before adding if you're using a shared object.

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Comment by bstansberry** [comment]
@ochaloup I want to merge this today as I'd like the upgrades in tonight. Can you send up a PR to address my minor test comments?

**Comment by bstansberry** [comment]
@ochaloup Sorry; my last comment was unclear. I meant please send up a separate PR as I'm merging this one.

**Comment by ochaloup** [comment]
thanks @bstansberry. @istudens fixed my ommisions in the PR https://github.com/wildfly/wildfly/pull/12059. Sorry for the errors here.

**Pull request merged**